model/docs/example-with-datasets.qmd

```diff
@@ -313,15 +313,15 @@ class WeekdayEffect(metaclass.RandomVariable):
         return None
 
     def sample(self, **kwargs):
-        ans = npro.sample(
+        and = npro.sample(
             name="weekday_effect",
             fn=npro.distributions.TruncatedNormal(
                 loc=1.0, scale=.5, low=0.1, high=10.0
                 ),
             sample_shape=(7,)
         )
 
-        return jnp.tile(ans, self.nweeks)[:self.len]
+        return jnp.tile(and, self.nweeks)[:self.len]
```

Comment:
I'm thinking, with this kind of "fix," the typos pre-commit hook may not be good enough. I'm afraid of it introducing bugs without us noticing. What do do you, @damonbayer and @dylanhmorris?

Reply:
Maybe, there's a way of correcting just the docstrings?

cc @natemcintosh

Reply:
There are a [couple of options](https://github.com/crate-ci/typos?tab=readme-ov-file#false-positives) for telling `typos` to ignore things. I've used file exclusion before to remove test data with false positives. 

But just docstrings might be harder. We could definitely hard code in `ans` as something to be ignored, but it's not the most ideal, and maybe one day we will want it to catch an `ans` that should have been a `and` in a README somewhere.

Overall, I'd say it might be worth considering moving typos from pre-commit to regular CI runs say once a week, or once a month? Then someone has to come and check the results though ...

Reply:
Catching up here. I wasn't aware that there was a plan to run something that did autocorrect on spelling for source code. My knee-jerk reaction is to be against any such thing, as it seems likely to cause far more problems than it solves.

Why not simply use non-correcting (i.e. just flagging) spell checkers in our local editors, or a report in CI that flags _potential_ spelling mistakes but does not change them?

Reply:
I feel similarly to Dylan. Big headache for minimal benefit. False negatives are not that bad, and false positives will be a frequency annoyance.

Reply:
We can simply change the input args for `typos` to not alter the code. See the bottom paragraph of [this page](https://github.com/crate-ci/typos/blob/master/docs/pre-commit.md).

In theory, we just need to add
`args: []` to the `typos` section of `.pre-commit-config.yaml`